src/Generation/EmulatorSupportGenerator.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+declare(strict_types=1);
+
+namespace Google\Generator\Generation;
+
+use Google\Auth\Credentials\InsecureCredentials;
+use Google\Generator\Ast\AST;
+use Google\Generator\Ast\Access;
+use Google\Generator\Ast\PhpDoc;
+use Google\Generator\Utils\Type;
+use \Grpc\ChannelCredentials;
+
+class EmulatorSupportGenerator
+{
+    /**
+     * Map of emulator support required clients and their expected env variables.
+     */
+    private static $emulatorSupportFixes = [
+        '\Google\Cloud\Spanner\Admin\Database\V1\Client\DatabaseAdminClient' => 'SPANNER_EMULATOR_HOST',
+        '\Google\Cloud\Spanner\Admin\Instance\V1\Client\InstanceAdminClient' => 'SPANNER_EMULATOR_HOST',
+        // Added for unittesting
+        '\Testing\Basic\Client\BasicClient' => 'BASIC_EMULATOR_HOST'
+    ];
```

Comment:
Is this supposed to work only for Spanner? I think this way could be used to generate the emulator vars for Datastore and others as well?

Reply:
For now, we only want it to work on Spanner, but yes we may expand it to others later.